### DISCUSSION TITLE
Can I deploy to railway.app?

### DISCUSSION THREAD

**Discussion by kstulgys** [pocketbase/pocketbase#923] [Q&A]
If so, how do I manage dev and prod environments? Can I somehow persist the data between builds /deploys?

**Comment by ganigeorgiev** [comment]
I don't have experience with railway but from a quick look at their [services documentation](https://docs.railway.app/develop/services) it doesn't look like they have option for mounting persistent storage like in fly.io. There is something what they call "persistent database" service but that will not work with PocketBase.

_**Edit:** The above is no longer true as Railway added support for volumes. See the comments below or the 1-click app in https://github.com/pocketbase/pocketbase/discussions/2784._

**Comment by GuilhermeAmado** [comment]
I have managed to deploy to Railway using this
https://github.com/GuilhermeAmado/pocketbase-docker

**Reply by GuilhermeAmado** [reply]
Oops forgot to mention it. You just have to use the Dockerfile in that repo.

**Reply by ganigeorgiev** [reply]
While it may be possible to start the PocketBase executable in railway, I still don't think you can actually use it as a host (unless you combine it with Litestream or something similar) because it doesn't seem that railway has persistent storage and you'll loose your `pb_data` on restart. Please correct me if that's not the case.

**Reply by YerayAlonso** [reply]
Indeed, you can deploy pocketbase in railway (done with [this](https://github.com/krushiraj/pocketbase-docker)) but you loose everything when you "Redeploy" (not when you just "Restart" the container)

![imagen](https://user-images.githubusercontent.com/1788228/207010879-e1d0a227-8114-453b-9e35-e13c3a25f9fa.png)

**Comment by SKumarSpace** [comment]
Railway released Volumes support - https://docs.railway.app/reference/volumes. This is now possible!
I wrote a tutorial for it here: https://www.shrey.com/blog/deploy-pocketbase-on-railway

**Reply by charbs-io** [reply]
Would you know how I can get pb_hooks directory to run?

**Reply by ganigeorgiev** [reply]
@charbs-io You can update your Dockerfile to copy the  local `pb_hooks`, `pb_migrations` etc. directories into the image with something like:

```dockerfile
COPY ./pb_hooks /pb_hooks
```

You can also find an alternative Dockerfile to the above example in https://pocketbase.io/docs/going-to-production/#using-docker (the volume in the docs example Dockerfile is expected to be mounted in `/pb/pb_data`).

**Reply by charbs-io** [reply]
Thanks @ganigeorgiev - i played around and got it working with this (for anyone interested).

```
FROM alpine:3 as downloader

ARG TARGETOS
ARG TARGETARCH
ARG VERSION=0.18.8

ENV BUILDX_ARCH="${TARGETOS:-linux}_${TARGETARCH:-amd64}"

# Install the dependencies
RUN apk add --no-cache \
    ca-certificates \
    unzip \
    wget \
    zip \
    zlib-dev

RUN wget https://github.com/pocketbase/pocketbase/releases/download/v${VERSION}/pocketbase_${VERSION}_${BUILDX_ARCH}.zip \
    && unzip pocketbase_${VERSION}_${BUILDX_ARCH}.zip \
    && chmod +x /pocketbase

COPY pb_hooks /pb_hooks
COPY pb_migrations /pb_migrations

FROM scratch

EXPOSE 8090

COPY --from=downloader /pocketbase /usr/local/bin/pocketbase
COPY --from=downloader /pb_hooks /pb_hooks
COPY --from=downloader /pb_migrations /pb_migrations

CMD ["/usr/local/bin/pocketbase", "serve", "--http=0.0.0.0:8090", "--dir=/pb_data", "--publicDir=/pb_public", "--hooksDir=/pb_hooks", "--migrationsDir=/pb_migrations"]
```

I have to say, Railway is an absolute breeze to work with - much easier than fly.io from my experience. 

I was even able to setup 2 environments (staging + production). My PB hooks are working, migrations are working. Domains were super easy to setup too (per environment). I feel like it's the perfect companion to PocketBase.

EDIT: @SKumarSpace - thanks again for the tutorial - it was very helpful.
